tensorboard/data/server/cargo/BUILD.bazel

```diff
@@ -0,0 +1,246 @@
+"""
+@generated
+cargo-raze generated Bazel file.
+
+DO NOT EDIT! Replaced on runs of cargo-raze
+"""
```

Comment:
Is this going to be added to `.gitattributes`?

Reply:
Also, which command generated this file?

Reply:
Thanks for the review!

> Is this going to be added to `.gitattributes`?

I considered marking it as binary in `.gitattributes`. I like it as is
because it’s human-readable, it only reflects our direct deps (not the
whole lockfile) and so won’t get too much churn, and its diffs are
human-readable and the blame on it will be useful. I’ll merge as is just
for expediency, but I’d be happy to change course if you’d like that or
if it becomes a problem later on.

> Also, which command generated this file?

This is generated with the workflow described in `server/DEVELOPMENT.md`
under the “Adding third-party dependencies” section. The specific
command that I always pull from my shell history is

```
(tb && rm -rf third_party/rust/ && cd tensorboard/data/server && cargo fetch && cargo raze)
```

(where `tb` is a `cd` to the repository root), which is basically an
abbreviation of the steps in that doc.

The file is new because `cargo-raze` now supports “Cargo workspaces”,
which are a way to have multiple packages in one repo. (Think of it kind
of like Lerna in the NPM world.) This means that the global namespace of
`//third_party/rust` is replaced with a per-`Cargo.toml` namespace. We
only have one `Cargo.toml`, so it’s just a move for us—but now that we
have workspace support I might actually want to take advantage of it to
separate out the gRPC code from the core.